dds/DCPS/security/AccessControl/XmlUtils.cpp

```diff
@@ -112,48 +114,268 @@ bool parse_bool(const XMLCh* in, bool& value)
   return true;
 }
 
+#define XMLDATETIME_HAS_GETEPOCH _XERCES_VERSION >= 30200
+#if !XMLDATETIME_HAS_GETEPOCH
+namespace {
+  bool parse_time_string(
+    const std::string& whole, size_t& pos, std::string& value, size_t width = std::string::npos)
+  {
+    const bool to_end = width == std::string::npos;
+    if (pos >= whole.size() || (!to_end && pos + width > whole.size())) {
+      return false;
+    }
+    try {
+      value = whole.substr(pos, width);
+    } catch (const std::out_of_range& e) {
```

Comment:
It doesn't seem possible to get an `out_of_range` here since `pos` is checked above.

Reply:
Yea, I added the check afterwards to get it behave how I wanted. I will go ahead and remove it.